CalibTracker/SiPixelConnectivity/src/SiPixelFedCablingMapBuilder.cc

```diff
@@ -30,34 +30,83 @@
+//SiPixelFedCablingMapBuilder::SiPixelFedCablingMapBuilder(const string & associatorName) : theAssociatorName(associatorName)
+SiPixelFedCablingMapBuilder::SiPixelFedCablingMapBuilder(const string fileName, 
+							 const bool phase1) : 
+  fileName_(fileName), phase1_(phase1)
 { }
 
 
-SiPixelFedCablingTree * SiPixelFedCablingMapBuilder::produce( const edm::EventSetup& setup)
-{
-  FEDNumbering fednum;
+SiPixelFedCablingTree * SiPixelFedCablingMapBuilder::produce( const edm::EventSetup& setup) {
+
+  //bool phase1_ = false;  
+  //bool phase1_ = true;  
+
+  cout<<" buildr "<<phase1_<<endl;
+
+  int MINSiPixelFEDID = 0;
+  int MAXSiPixelFEDID = 39;
+  if(phase1_) {
+    // bpix 1200-1239, fpix 1240-1255
+    MINSiPixelFEDID = 1200;
+    MAXSiPixelFEDID = 1255;
+  }
+
+  //FEDNumbering fednum;
+  // fed range should come from input ot automatic from geometry
 //  TRange<int> fedIds(FEDNumbering::MINSiPixelFEDID, FEDNumbering::MAXSiPixelFEDID);
-  TRange<int> fedIds(0,39);
-  edm::LogInfo("SiPixelFedCablingMapBuilder")<<"pixel fedid range: "<<fedIds;
 
+  TRange<int> fedIds(MINSiPixelFEDID,MAXSiPixelFEDID);
+  edm::LogInfo("SiPixelFedCablingMapBuilder")<<"pixel fedid range: "<<fedIds;
 
-  edm::ESHandle<PixelToFEDAssociate> associator;
-  setup.get<TrackerDigiGeometryRecord>().get(theAssociatorName,associator);
-//   PixelToFEDAssociate * associator = new PixelToLNKAssociateFromAscii("pixelToLNK.ascii");
+  // in the constrcuctor init() is called which reads the ascii file and makes 
+  // the map roc<->link 
+  // what is this junk?
+  //edm::ESHandle<PixelToFEDAssociate> associator;
+  //setup.get<TrackerDigiGeometryRecord>().get(theAssociatorName,associator);
 
+  //PixelToFEDAssociate * associator = new PixelToLNKAssociateFromAscii("pixelToLNK.ascii",phase1_);
+  PixelToFEDAssociate * associator = new PixelToLNKAssociateFromAscii(fileName_,phase1_);
   
   const PixelToFEDAssociate & name2fed = *associator; 
   
   string version = name2fed.version();
   SiPixelFedCablingTree * result = new SiPixelFedCablingTree(version);
+  edm::LogInfo(" version ")<<version<<endl;
 
-
-  LogDebug("read tracker geometry...");
+  // Access geometry
+  edm::LogInfo("read tracker geometry...");
   edm::ESHandle<TrackerGeometry> pDD;
   setup.get<TrackerDigiGeometryRecord>().get( pDD );
-  LogDebug("tracker geometry read")<<"There are: "<<  pDD->dets().size() <<" detectors";
+  edm::LogInfo("tracker geometry read")<<"There are: "<<  pDD->dets().size() <<" detectors";
+
+  // Test new TrackerGeometry features
+  cout << "Test of TrackerGeometry::isThere";
+  cout  << " is there PixelBarrel: " << pDD->isThere(GeomDetEnumerators::PixelBarrel);
+  cout  << " is there PixelEndcap: " << pDD->isThere(GeomDetEnumerators::PixelEndcap);
+  cout  << " is there P1PXB: " << pDD->isThere(GeomDetEnumerators::P1PXB);
+  cout  << " is there P1PXEC: " << pDD->isThere(GeomDetEnumerators::P1PXEC);
+  cout  << endl;
+
```

Comment:
@dkotlins as reported by @mmusich in https://github.com/cms-sw/cmssw/pull/12757/files#r47640725 you should use `MessageLogger` API.